lib/benchee/system.ex

```diff
@@ -164,4 +172,12 @@ defmodule Benchee.System do
       output
     end
   end
+
+  defp all_protocols_consolidated?() do
+    path = :code.lib_dir(:elixir, :ebin)
+
+    [path]
+    |> Protocol.extract_protocols()
```

Comment:
I have no idea how this works :D

Reply:
It takes a list of paths to directories full of BEAM files and tells you which protocols have been defined in those directories. It nicely returns the list of atoms corresponding to the protocol name. Very helpful when combined with `:code.lib_dir/2` to get the top level directory for the Elixir BEAM files. I used this as a proxy since if those aren't consolidated, then there's no way any of your other protocols are consolidated. In theory it's still possible for a protocol to have been defined and not consolidated, but that would emit a warning (like if you're defining a protocol in a test body or something).